src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -217,7 +222,20 @@ protected Object invokeStep(StepDescriptor d, Object args) {
      * @param args The arguments passed to the step.
      */
     protected Object invokeStep(StepDescriptor d, String name, Object args) {
-        final NamedArgsAndClosure ps = parseArgs(args, d);
+        Set<String> interpolatedStrings = null;
+        if (args instanceof NamedArgsAndClosure) {
+            interpolatedStrings = ((NamedArgsAndClosure) args).interpolatedStrings;
+        } else if (args instanceof Object[]) {
+            Object[] array = (Object[]) args;
+            if (array.length > 0 && array[array.length - 1] instanceof InterpolatedUninstantiatedDescribable) {
+                interpolatedStrings = ((InterpolatedUninstantiatedDescribable) array[array.length - 1]).getInterpolatedStrings();
```

Comment:
Looking at this code again, I don't think I need to make sure I'm grabbing the last element of the array and can just grab the first element?

Reply:
I'm not sure, maybe you need to check every argument? For something like `monomorphWithSymbolStep(monomorphSymbol('foo', 'bar'))`, is `args` here an `Object[]` with an `InterpolatedUninstantiatedDescribable` as the only element (corresponding to `monomorphSymbol('foo', 'bar')`), a `NamedArgsAndClosure`, or just the `InterpolatedUninstantiatedDescribable` directly?

Also, could this logic be embedded in `parseArgs` instead, since it already processes each argument?

Reply:
Its an `Object[]' where the first element is an 'InterpolatedUninstantiatedDescribable`

Reply:
Checking all elements of Object[] in 70c4c223dd4d08ca065fb9f8b0e41bb3ccc07c6d